tools/gluten-te/ubuntu/dockerfile-buildenv

```diff
@@ -3,14 +3,21 @@ ARG OS_IMAGE
 FROM $OS_IMAGE AS gluten-buildenv
 MAINTAINER Hongze Zhang<hongze.zhang@intel.com>
 
-SHELL ["/bin/bash", "-c"]
+SHELL ["/bin/bash", "-l", "-c"]
+ENTRYPOINT ["/bin/bash", "-l", "-c"]
+CMD ["/bin/bash"]
 
 # REQUIRED PROXYS: APT, WGET, GIT, MAVEN (also Maven mirror)
 ARG HTTP_PROXY_HOST
 ARG HTTP_PROXY_PORT
 
-ENV http_proxy=${HTTP_PROXY_HOST:+"http://$HTTP_PROXY_HOST:$HTTP_PROXY_PORT"}
-ENV https_proxy=${HTTP_PROXY_HOST:+"http://$HTTP_PROXY_HOST:$HTTP_PROXY_PORT"}
+# Sometimes ENV a=b won't work when the shell is not docker-default
+RUN echo "export http_proxy=${HTTP_PROXY_HOST:+http://$HTTP_PROXY_HOST:$HTTP_PROXY_PORT}" >> /etc/profile.d/99-gluten.sh
+RUN echo "export https_proxy=${HTTP_PROXY_HOST:+http://$HTTP_PROXY_HOST:$HTTP_PROXY_PORT}" >> /etc/profile.d/99-gluten.sh
+RUN echo "export no_proxy=localhost,127.0.0.1" >> /etc/profile.d/99-gluten.sh
+RUN echo "export HTTP_PROXY=${HTTP_PROXY_HOST:+http://$HTTP_PROXY_HOST:$HTTP_PROXY_PORT}" >> /etc/profile.d/99-gluten.sh
+RUN echo "export HTTPS_PROXY=${HTTP_PROXY_HOST:+http://$HTTP_PROXY_HOST:$HTTP_PROXY_PORT}" >> /etc/profile.d/99-gluten.sh
+RUN echo "export NO_PROXY=localhost,127.0.0.1" >> /etc/profile.d/99-gluten.sh
```

Comment:
This may not work in the case of non-login shell. I'll raise another PR to fix that.